packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -0,0 +1,150 @@
+  /// the output to [outputPath].
+  ///
+  /// Returns `true` if successful and `false` otherwise.
+  Future<bool> transformAsset({
+    required File asset,
+    required String outputPath,
+    required String workingDirectory,
+    required List<AssetTransformerEntry> transformerEntries,
+  }) async {
+
+    String getTempFilePath() => '${_random.nextInt(1<<32)}${_fileSystem.path.extension(asset.path)}';
+
+    File tempInputFile = _fileSystem.systemTempDirectory.childFile(getTempFilePath());
+    await asset.copy(tempInputFile.path);
+    File tempOutputFile = _fileSystem.systemTempDirectory.childFile(getTempFilePath());
+
+    try {
+      for (final (int i, AssetTransformerEntry transformer) in transformerEntries.indexed) {
+        final _AssetTransformerFailure? transformerFailure = await _applyTransformer(
+          asset: tempInputFile,
+          output: tempOutputFile,
+          transformer: transformer,
+          workingDirectory: workingDirectory,
+        );
+
+        if (transformerFailure != null) {
+          _logger.printError(
+            'User-defined transformation of asset "${asset.path}" failed.\n'
+            '${transformerFailure.message}',
+          );
+          break;
+        }
+
+        ErrorHandlingFileSystem.deleteIfExists(tempInputFile);
+        if (i == transformerEntries.length - 1) {
+          await tempOutputFile.copy(outputPath);
+        } else {
+          tempInputFile = tempOutputFile;
+          tempOutputFile = _fileSystem.systemTempDirectory.childFile(getTempFilePath());
+        }
+      }
+    } finally {
+      ErrorHandlingFileSystem.deleteIfExists(tempInputFile);
+      ErrorHandlingFileSystem.deleteIfExists(tempOutputFile);
+    }
+
+    return true;
+  }
+
+  Future<_AssetTransformerFailure?> _applyTransformer({
+    required File asset,
+    required File output,
+    required AssetTransformerEntry transformer,
+    required String workingDirectory,
+  }) async {
+    final List<String> transformerArguments = <String>[
+      '--input=${asset.absolute.path}',
+      '--output=${output.absolute.path}',
+      ...?transformer.args,
+    ];
+
+    final List<String> command = <String>[
+      _dartBinary,
+      'run',
+      transformer.package,
+      ...transformerArguments,
+    ];
+
+    final ProcessResult result = await _processManager.run(
+      command,
+      workingDirectory: workingDirectory,
+    );
+    final String stdout = result.stdout as String;
+    final String stderr = result.stderr as String;
+
+    if (result.exitCode != 0) {
+      return _AssetTransformerFailure(
+        'Transformer process terminated with non-zero exit code: ${result.exitCode}\n'
+        'Transformer package: ${transformer.package}\n'
+        'Full command: ${command.join(' ')}\n'
+        'stdout:\n$stdout\n'
+        'stderr:\n$stderr'
+      );
+    }
+
+    if (!await _fileSystem.file(output).exists()) {
```

Comment:
nit. I think we can just do `existsSync()`?

Reply:
good catch